packages/plugins/plugin-table/src/components/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { ChatText } from '@phosphor-icons/react';
+import { createContext } from '@radix-ui/react-context';
+import React, { type PropsWithChildren } from 'react';
+
+import {
+  DensityProvider,
+  ElevationProvider,
+  Toolbar as NaturalToolbar,
+  type ThemedClassName,
+  useTranslation,
+} from '@dxos/react-ui';
+
+import { ToolbarButton, ToolbarSeparator } from './common';
+import { TABLE_PLUGIN } from '../../meta';
+
+//
+// Root
+//
+
+// TODO(Zan): This should become a union later.
+export type ToolbarAction = { type: 'comment' };
+
+export type ToolbarActionType = ToolbarAction['type'];
+
+export type ToolbarActionHandler = (action: ToolbarAction) => void;
+
+export type ToolbarProps = ThemedClassName<
+  PropsWithChildren<{
+    onAction?: ToolbarActionHandler;
+  }>
+>;
+
+const [ToolbarContextProvider, useToolbarContext] = createContext<ToolbarProps>('Toolbar');
+
+const ToolbarRoot = ({ children, onAction, classNames }: ToolbarProps) => {
+  return (
+    <ToolbarContextProvider onAction={onAction}>
+      <DensityProvider density='fine'>
+        <ElevationProvider elevation='chrome'>
```

Comment:
Do either of these do anything still? Maybe so, just considering a refactor to do this with CSS rather than providers…